src/helm/benchmark/scenarios/cleva_scenario.py

```diff
@@ -0,0 +1,1491 @@
+CLEVA_DATA_PATH = "benchmark_output/scenarios/cleva"
+
+
+@dataclass(frozen=True)
+class PromptSetting:
+    """
+    Specifies prompt-related settings for AdapterSpec.
+    """
+
+    # Method of adaptation
+    method: str = ""
+
+    # Prepend all prompts with this string.
+    global_prefix: str = ""
+
+    # Prompt starts with instructions
+    instructions: str = ""
+
+    # What goes before the input
+    input_prefix: str = ""
+
+    # What goes after the input
+    input_suffix: str = "\n"
+
+    # What goes before the input (for multiple choice)
+    reference_prefix: str = "A. "
+
+    # What goes before the input (for multiple choice)
+    reference_suffix: str = "\n"
+
+    # What goes before the output
+    output_prefix: str = ""
+
+    # What goes after the output
+    output_suffix: str = "\n"
+
+    # What goes between instruction and in-context example blocks in the constructed prompt
+    instance_prefix: str = "\n"
+
+
+class Converter:
+    """
+    Convert samples in CLEVA format to HELM instances according to CLEVA prompt template standard.
+    """
+
+    def transform(self, data: Dict[str, Any], templates: Dict[str, Any], split: str) -> Instance:
+        """Convert a data point in CLEVA format to a HELM instance according to a given CLEVA prompt template."""
+        transformed_data = self._apply_all(copy.deepcopy(data), templates)
+
+        text = transformed_data["input"]
+        if "choices" in transformed_data:
+            # This is a multiple-choice task
+            references: List[Reference] = [
+                Reference(Output(text=text), tags=[CORRECT_TAG] if idx in transformed_data["label"] else [])
+                for idx, text in enumerate(transformed_data["choices"])
+            ]
+        else:
+            # This is a generation task
+            correct_answer = transformed_data["label"]
+            references = [Reference(Output(text=answer), tags=[CORRECT_TAG]) for answer in correct_answer]
+
+        instance = Instance(
+            input=Input(text=text),
+            references=references,
+            split=split,
+        )
+        return instance
+
+    def transform_code(self, data: Dict[str, Any], templates: Dict[str, Any], split: str) -> CodeInstance:
+        """
+        Similar to transform method above, transform_code converts a data point in code synthesis scenario in CLEVA
+        to a HELM CodeInstance according to a given CLEVA prompt template.
+        """
+
+        data["prompt"] = templates["input"].format(**data)
+        instance = CodeInstance(
+            input=Input(text=data["prompt"]),
+            references=[
+                CodeReference(
+                    output=Output(text=data["canonical_solution"]),
+                    test_cases=data,
+                    tags=[CORRECT_TAG],
+                )
+            ],
+            split=split,
+        )
+        return instance
+
+    def _apply_all(self, data: dict, templates: dict) -> dict:
```

Comment:
Sounds like templates should be of type `templates: Dict[str, Optional[Template]]` (see below for the definition of `Template`)